src/Profile.php

```diff
@@ -778,6 +778,363 @@ public function showForm($ID, array $options = [])
         return true;
     }
 
+    /**
+     * Get all rights to display for a specific form and interface.
+     *
+     * This is only used for GLPI core rights and not rights added by plugins.
+     *
+     * @param string $form The tab/form name
+     * @phpstan-param non-empty-string $form
+     * @param string $interface The interface name
+     * @phpstan-param 'central'|'helpdesk' $interface
+     * @return array
+     * @phpstan-return array<string, array{
+     *     rights?: array{}, itemtype?: class-string<CommonDBTM>, label: string, field: string, group: string, scope: string
+     * }>
+     * @internal BC not guaranteed. Only public so it can be used in tests to ensure search options are made for all rights.
+     */
+    public static function getRightsForForm(string $form, string $interface): array
```

Comment:
Usage of this method could probably be simplified in order to not have to filter results.
```suggestion
    public static function getRightsSpecs(?string $group = null): array
```

`$group` could be something like '{$interface}/{$form}(/{$section})', for instance:
- `helpdesk/tracking`
- `helpdesk/tools`
- `central/assets`
- `central/tools/general`
- `central/tools/projects`

With this, usage in profile form would be simple as:
```php
$this->displayRightsChoiceMatrix(self::getRightsSpecs('central/tools/projects'), $matrix_options);
```
and usage from tests could be simple as 
:
```php
$rights = \Profile::getRightsSpecs(); // No $group param passed => get all possible rights
foreach ($rights as $right) {
    ...
}
```

Reply:
I'm not sure about `helpdesk/tracking` and `central/tools/general` style for getting rights instead of parameters. I don't think it really fits with anything else in the code. If anything, I think another parameter should be added for the section/group and do the filtering within the function itself and change the default group from `null` to `general`. Then, `null` would return all sections/groups for a specific form and interface.

I'd like to have other opinions on this though.